src/components/upload-history/index.js

```diff
@@ -0,0 +1,41 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import FileSaver from "file-saver";
+
+const mapStateToProps = state => ({
+  items: state.upload.history
+});
+
+const mapDispatchToProps = dispatch => ({
+});
+
+function generateUploadHistoryFileFn(items) {
```

Comment:
this should go into an epic (https://github.com/redux-observable/redux-observable)

So basically the flow is this:
1. user clicks on the button, that fires a redux action called DOWNLOAD_HANDLES (or whatever)
2. the epic "observes" / listens to the redux action and executes the "side effect" code

The purpose of this is to extract all the business logic out of the components.

Ex:
https://github.com/oysterprotocol/webinterface/blob/9432b7344f2a7d97526d253b08f678c959b2654f/src/redux/epics/download-epic.js#L14